lib/utils/filter_samples.py

```diff
@@ -0,0 +1,127 @@
+# Primary file for filtering samples workflows
+from utils.re_utils import execute_query
+from utils.parsing_and_formatting import (
+    parse_input,
+    parse_field,
+    parse_values,
+    parse_comparison_operator,
+    parse_logical_operator,
+    field_value_formatting
+)
+
+SAMPLE_NODE_COLLECTION = "samples_nodes"
+SAMPLE_SAMPLE_COLLECTION = "samples_sample"
+
+# AQL = Arango Query Languages
+# double curly braces "{{}}" are used to create string literal curly braces "{}"
+AQL_query_template = f"""
+let version_ids = (for sample_id in @sample_ids
+    let doc = DOCUMENT({SAMPLE_SAMPLE_COLLECTION}, sample_id.id)
+    RETURN {{
+        'id': doc.id,
+        'version_id': doc.vers[sample_id.version - 1],
+        'version': sample_id.version
+    }}
+)
+
+let node_metas = (for version_id in version_ids
+    for node in {SAMPLE_NODE_COLLECTION}
+        FILTER node.id == version_id.id and node.uuidver == version_id.version_id
+        LIMIT @num_sample_ids
+        let meta_objs_values = (for meta in node.cmeta
+            RETURN {{ [ meta.ok ]: {{ [ meta.k ]: meta.v }} }}
+        )
+        RETURN {{
+            "id": node.id,
+            "version": node.ver,
+            "meta": APPLY("MERGE_RECURSIVE", meta_objs_values)
+        }}
+)
+for node in node_metas
+    FILTER """
+
+
+class SampleFilterer():
+    '''
+    '''
+    def __init__(cls, ctx, re_api_url, sample_service):
+        cls.re_api_url = re_api_url
+        cls.sample_service = sample_service
+        cls.re_admin_token = ctx.get('token')  # needs to be a env/config variable
```

Comment:
probs add a config variable for this to `deploy.cfg`?

Reply:
good call

Reply:
turns out this is unnecessary, we'll just have to define once deployed